### ISSUE TITLE
Navigating dialog boxes or applications like on a web page

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3676]
**Reported by heikofolkerts on 2013-11-30 19:34**
Many applications are quite inaccessible because they have no good keyboard control. Many dialogs in such applications have no tab order etc.

If NVDA can recognize the objects within the application it would be very helpful if one could navigate through the controls like one navigates within a web page. E.g. one could jump to all radio buttons in a dialog by pressing r multiple times. The order should be derived from the object tree in deep search mode so that objects contained in a parent object will be visited together. This may not be in a good order but if the application has no keyboard control this is far better.

The requested feature should place the navigator on the next requested object so that the user can read and or activate it. The quick navigation keys can be very close to those in web pages and adding e.g. t for text boxes. If possible the dialog could also be rendered in a kind of virtual buffer so that the user can read the contents by arrowing down.

The requested mode should respect overlay classes so that it is possible to map unrecognized windows classes to known ones.

**Blocking #4718**

**Comment by bhavyashah** [comment]
CC @leonardder and @derekriemer for general thoughts on this pretty interesting idea and @josephsl to discuss this feature request's impact on Windows 10 UWP apps.

**Comment by derekriemer** [comment]
There is more chance of this happening now that mick has implemented edge support, and we have more idea of how to do this.

**Comment by leonardder** [comment]
It would be good to have up to date thoughts from @@michaeldcurran and @jcsteh about this subject.

Note that this feature suggestion has many aspects that should be considered. For example, how to deal with display model information? I've been working on a system that integrates display model based object nicely into the nVDA objects tree, so that could help.

**Comment by Adriani90** [comment]
I suggest closing this one in favour of #7952 which contains more in depth discussions and even a small global plugin which simplifies object navigation.

**Comment by Adriani90** [comment]
Closing in favor of #7759. Let's continue discussions on that issue.
